[#2] 프론트엔드 애플리케이션 개발 환경 구축

woobuntu opened the pull request:
- Dockerfile: 현재 계획상 운영 환경에서는 S3 + CloudFront조합으로 배포할 것이므로, production 스테이지는 없을 예정. 다만, 추후 build 스테이지는 추가될 수 있음.
- .env: docker-compose 커맨드를 축약해서 사용하기 위해 만든 환경변수

Commit:
프론트엔드 애플리케이션 개발 환경 구축

- Dockerfile: 현재 계획상 운영 환경에서는 S3 + CloudFront조합으로 배포할 것이므로, production 스테이지는 없을 예정. 다만, 추후 build 스테이지는 추가될 수 있음.
- .env: docker-compose 커맨드를 축약해서 사용하기 위해 만든 환경변수

Commit:
Merge branch 'main' of https://github.com/f-lab-edu/double-entry-book-keeping into feature/2

woobuntu committed:
백엔드 애플리케이션 개발 환경 구축

프론트엔드와는 달리 백엔드는 개발 환경과 운영 환경 모두 node 컨테이너에서 실행되므로 base 스테이지를 정의하여 재사용

woobuntu committed:
데이터베이스 컨테이너 추가

- 데이터베이스 컨테이너 -> 서버 컨테이너 -> 클라이언트 컨테이너 순의 종속성 설정
- 데이터베이스 컨테이너는 오직 서버 컨테이너에만 노출하기 위해 private network 설정

woobuntu committed:
prisma 설정

- 서버에서는 prisma orm을 사용
- 서버 도커 파일에서 prisma 관련 내용 추가
- 도커 컴포즈에서 prisma bind mount + 커넥션 url 환경변수 파일로 전달

choru-k commented on .env:
```diff
@@ -0,0 +1,3 @@
+COMPOSE_PATH_SEPARATOR=;
+COMPOSE_FILE=docker-compose.yml;docker-compose-dev.yml
+COMPOSE_PROJECT_NAME=double-entry-book-keeping
```
마지막 개행이 있는편이 좋을 것 같습니다.

woobuntu replied:
@g6ling 말씀주신 대로 마지막에 개행을 추가하였습니다! 그런데 env파일에서 마지막 개행이 있는 것의 의미가 무엇일까요...?

woobuntu committed:
chore: .env 마지막 개행 추가

woobuntu committed:
Merge branch 'feature/2' of https://github.com/f-lab-edu/double-entry-book-keeping into feature/5

woobuntu committed:
Merge branch 'feature/5' of https://github.com/f-lab-edu/double-entry-book-keeping into feature/7

woobuntu committed:
데이터베이스 컨테이너 헬스 체크 추가

- https://github.com/docker-library/healthcheck/tree/master/mysql 로부터 Dockerfile과 docker-healtcheck 참조
- docker-healthcheck의 10번째 줄은 원래 host="$(hostname --ip-address || echo '127.0.0.1')"였으나 hostname 커맨드가 없어서 수정
- healtcheck interval, timeout, retries 등은 임의 설정

woobuntu committed:
서버 컨테이너 헬스 체크 추가

- nestjs의 terminus 패키지로 헬스 체크 엔드포인트 추가
- 해당 헬스 체크에서 데이터베이스가 연결되었는지, 즉 디펜던시 체크도 가능한 부분이라 서버 이미지에 달리 디펜던시 체크를 추가하지 않았다.
- 또한, 헬스 체크에 디펜던시 체크가 이미 존재하고 있어 서버 컨테이너의 depends_on 구문도 삭제하였다.

woobuntu committed:
Merge pull request #8 from f-lab-edu/feature/7

[#7] 개발환경에 데이터베이스 연동 + prisma orm 설정

woobuntu committed:
Merge pull request #6 from f-lab-edu/feature/5

[#5] 백엔드 애플리케이션 개발 환경 구축
